### PR TITLE
Add conversation service

### PR DESCRIPTION
**Pull request by nicksellen**

### CODE REVIEW

**Comment by nicksellen** [comment]
Ok, I added tests! Not the most dazzlingly complete tests in the world, but maybe good enough for now?

If I removed the demo UI, would you be happy to merge to master @tiltec? - it would just have an extra service that would not be used anywhere.

I would open a new PR with the demo UI that could be the start of the UI work....

File: client/app/services/conversation/socket.service.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        this.subscribers.splice(idx, 1);
+      }
+    };
+  }
+
+  /* Connect if not already connected */
+  connect() {
+    if (this.socket) return;
+    this.socket = new ReconnectingWebSocket(
+      "ws://" + window.location.host + "/api/ws",
```
Comment (tiltec) [inline client/app/services/conversation/socket.service.js]: Will this also work over HTTPS? I heard something of `wss://`

**Comment by tiltec** [comment]
Looks solid to me, I would merge the service!

You could even leave the `conversation(groupId)` method in the groupService code (but add a little test).

**Comment by nicksellen** [comment]
Oh yeah need to check for htts then use wss.

I would remove it from the group detail entirely as i think depending on ui
it might not end up there.

**Comment by nicksellen** [comment]
> You could even leave the conversation(groupId) method in the groupService code (but add a little test).

Ah, I didn't read it properly first time. I understand now, and did that.

**Comment by nicksellen** [comment]
Assuming the tests pass, then this is good to go!

**Pull request merged**